src/main/java/net/mcreator/element/types/Item.java

```diff
@@ -220,14 +223,105 @@ public List<StateEntry> getModels() {
 					model.stateMap.put(prop, value);
 			});
 
-			// only add this state if at least one supported property is present
-			if (!model.stateMap.isEmpty())
+			// only add this state if not duplicate and at least one supported property is present
+			if (!models.contains(model) && !model.stateMap.isEmpty()) {

```

Comment:
Why do we need to reduce duplicates and did not have to so far?

Reply:
We actually filter out duplicates in UI, not sure why this method wasn't filtering them so far as well.

Reply:
If we filter in UI, why also here?

And since this is not really related to the porting, it should not be part of this PR to keep changes to minimum.

If it works -> don't fix it

Reply:
It works with this filtering, I didn't really check how the tree would be constructed if there are duplicates present.
> If we filter in UI, why also here?

UI filters them when removing a property results into states with duplicate `StateMap`s, here we are removing some properties too but not cleaning up resulting duplicates.

Reply:
If it is not needed, I would avoid that. Comparing the equals and hashCode methods operating on Map are not ideal either so if this can be designed in a way we don't need that, it is better that way

Reply:
Without `equals()` this would need to use `stream()` and directly compare state maps of the entries processed earlier.

Reply:
You don't need equals and hashCode if you don't remove duplicates here